core/src/main/java/org/apache/carbondata/core/util/CarbonTestUtil.java

```diff
@@ -161,6 +166,50 @@ public static Boolean checkForLocalDictionary(
     return isLocalDictionaryGenerated;
   }
 
+  public static int getSegmentFileCount(String tableName) throws IOException {
+    CarbonTable carbonTable = CarbonMetadata.getInstance().getCarbonTable(tableName);
+    CarbonFile segmentsFolder = FileFactory
+        .getCarbonFile(CarbonTablePath.getSegmentFilesLocation(carbonTable.getTablePath()));
+    assert (segmentsFolder.isFileExist());
+    return segmentsFolder.listFiles(true).size();
+  }
+
+  public static int getIndexFileCount(String tableName,
+      String segment, String extension) throws IOException {
+    if (extension == null) {
+      extension = CarbonTablePath.INDEX_FILE_EXT;
+    }
+    CarbonTable table = CarbonMetadata.getInstance().getCarbonTable(tableName);
+    String path = CarbonTablePath
+        .getSegmentPath(table.getAbsoluteTableIdentifier().getTablePath(), segment);
+    CarbonFile[] carbonFiles;
```

Comment:
Can optimize as below:
```suggestion
     List<CarbonFile> carbonFiles;
    boolean recursive = false;
    if (table.isHivePartitionTable()) {
      path = table.getAbsoluteTableIdentifier().getTablePath();
      recursive = true;
    }
    carbonFiles = FileFactory.getCarbonFile(path).listFiles(recursive,
        file -> file.getName().endsWith(CarbonTablePath.INDEX_FILE_EXT) || file.getName()
            .endsWith(CarbonTablePath.MERGE_INDEX_FILE_EXT));
    CarbonFile[] validIndexFiles = (CarbonFile[]) SegmentFileStore
        .getValidCarbonIndexFiles(carbonFiles.toArray(new CarbonFile[carbonFiles.size()]));
```